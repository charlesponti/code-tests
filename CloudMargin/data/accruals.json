[{
    "DATE": "2017-01-01",
    "RATE": 0.5,
    "BALANCE": 10000,
    "ACCRUAL": 5000
}, {
    "DATE": "2017-01-02",
    "RATE": 0.5,
    "BALANCE": 12000,
    "ACCRUAL": 6000
}, {
    "DATE": "2017-01-03",
    "RATE": 0.5,
    "BALANCE": 0,
    "ACCRUAL": 0
}, {
    "DATE": "2017-01-04",
    "RATE": 0.25,
    "BALANCE": 10000,
    "ACCRUAL": 2500
}, {
    "DATE": "2017-01-05",
    "RATE": 0.5,
    "BALANCE": 10000,
    "ACCRUAL": 5000
}]
