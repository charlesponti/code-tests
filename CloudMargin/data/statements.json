[{
	"AMENDMENT": -300,
	"START_DATE": "2017-01-01",
	"END_DATE": "2017-01-31",
	"TOTAL": 18200
}, {
	"AMENDMENT": 0,
	"START_DATE": "2017-02-01",
	"END_DATE": "2017-02-28",
	"TOTAL" : 0
}, {
	"AMENDMENT": 555.55  ,
	"START_DATE": "2017-01-03",
	"END_DATE": "2017-03-31",
	"TOTAL": 555.55
}]