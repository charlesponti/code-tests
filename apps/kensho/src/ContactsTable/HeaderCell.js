import * as React from 'react'
import PropTypes from 'prop-types'
import { Button, InputGroup, Popover } from '@blueprintjs/core'

export default function HeaderCell(props) {
  return (
    <th className="headerCell">
      {props.name}
      <Popover>
        <Button icon="filter" minimal />
        <InputGroup
          value={props.filter}
          onChange={props.onFilter}
          placeholder="Filter Rows"
          leftIcon="filter"
          autoFocus
        />
      </Popover>
    </th>
  )
}

HeaderCell.propTypes = {
  name: PropTypes.string,
  filter: PropTypes.string,
  onFilter: PropTypes.func,
  type: PropTypes.oneOf(['category', 'number', 'string']),
}
