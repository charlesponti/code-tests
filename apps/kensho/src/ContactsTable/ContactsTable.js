import * as React from 'react'
import PropTypes from 'prop-types'
import { HTMLTable } from '@blueprintjs/core'
import HeaderCell from './HeaderCell'

export default class ContactsTable extends React.Component {
  static propTypes = {
    columns: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        key: PropTypes.string,
        type: PropTypes.oneOf(['category', 'number', 'string']),
      })
    ),
    contacts: PropTypes.arrayOf(PropTypes.object),
  }

  constructor(props) {
    super(props);

    this.state = {
      // Construct filters based on the names of the columns available
      filters: this.props.columns.reduce(
        (a, b) => ({ ...a, [b.key]: '' }),
        {}
      )
    }
  }

  handleFilter = column => event => {
    this.setState({
      filters: {
        ...this.state.filters,
        [column]: event.target.value
      }
    })
  }

  doesContactContainFilter(contact, filterName) {
    const value = this.state.filters[filterName]

    return value === ''
      ? true
      : (contact[filterName] + '').toUpperCase().includes(value.toUpperCase())
  }

  filteredContacts() {
    const { contacts } = this.props

    const filtered = contacts.filter(contact => {
      const map = Object
        .keys(this.state.filters)
        .map((filterName) => this.doesContactContainFilter(contact, filterName))

      // Only return contact if all filters match
      return !map.includes(false)
    })

    return filtered
  }

  removeFilter(filter) {
    return () => this.setState({
      filters: {
        ...this.state.filters,
        [filter]: ''
      }
    })
  }

  render() {
    const { filters } = this.state
    const { columns } = this.props

    return (
      <HTMLTable striped interactive className="table is-bordered is-striped is-narrow is-hoverable is-fullwidth">
        <thead>
          <tr>
            <td colSpan={columns.length}>
              {Object.keys(filters).map(f => filters[f] !== '').includes(true) && (
                <div style={{ fontWeight: 'bold', fontSize: '2rem', marginBottom: '0.5rem', borderBottom: '1px solid black' }}>
                  Current Filters
                </div>
              )}
              {Object.keys(filters).map((filter) => (
                filters[filter] !== '' ?
                  (
                    <div key={filter} style={{ fontSize: '1.5rem', marginBottom: '0.5rem' }}>
                      <span style={{ marginRight: '1rem' }}>
                        <b>{filter.toUpperCase()}</b> : {filters[filter] === '' ? 'N/A' : filters[filter]}
                      </span>
                      {filters[filter] !== '' && (
                        <button
                          className="button is-danger"
                          onClick={this.removeFilter(filter)}
                        >
                          Remove
                          </button>
                      )
                      }
                    </div>
                  )
                  : null
              ))}
            </td>
          </tr>
          <tr>
            {columns.map((col, index) => (
              <HeaderCell
                filter={filters[col.key]}
                onFilter={this.handleFilter(col.key)}
                key={index}
                {...col}
              />
            ))}
          </tr>
        </thead>
        <tbody>
          {this.filteredContacts().map(contact => (
            <tr key={contact.phone}>
              {columns.map(column => <td key={column.key}>{contact[column.key]}</td>)}
            </tr>
          ))}
        </tbody>
      </HTMLTable>
    )
  }
}
