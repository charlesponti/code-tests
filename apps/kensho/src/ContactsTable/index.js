import * as React from 'react'

import { getContacts } from '../api'

import ContactsTable from './ContactsTable'

export default class ContactsTableWrapper extends React.Component {
  state = {
    columns: null,
    contacts: null,
    isLoading: true,
  }

  componentDidMount() {
    this.fetchContacts()
  }

  fetchContacts() {
    getContacts().then(data => {
      this.setState({ isLoading: false, contacts: data.rows, columns: data.metadata.columns })
    })
  }

  render() {
    const { columns, contacts, isLoading } = this.state
    return isLoading
      ? (
        <div className="bp3-progress-bar bp3-intent-success">
          Loading
          <div className="bp3-progress-meter" style={{ width: '75%' }}></div>
        </div>
      )
      : <ContactsTable contacts={contacts} columns={columns} />
  }
}
