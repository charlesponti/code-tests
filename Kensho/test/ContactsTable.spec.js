// Link.react.test.js
import React from 'react';
import ContactsTable from '../src/ContactsTable/ContactsTable';
import { shallow } from 'enzyme'
import data from '../src/data'

const mocks = [{
  "age": 35,
  "city": "Maquanzhen",
  "name": "Gwen Garrick",
  "phone": "776-878-8338"
}, {
  "age": 40,
  "city": "General Lavalle",
  "name": "Carmella Brunker",
  "phone": "922-694-8628"
}, {
  "age": 2,
  "city": "Xinba",
  "name": "Abbot Delahunt",
  "phone": "327-542-4874"
}, {
  "age": 7,
  "city": "Mlaka pri Kranju",
  "name": "Lucine Stratz",
  "phone": "422-562-2609"
}, {
  "age": 98,
  "city": "Kamenický Šenov",
  "name": "Ethe Tomankiewicz",
  "phone": "580-701-9078"
}]

test('Displays data', () => {
  const tree = shallow(
    <ContactsTable columns={data.metadata.columns} contacts={mocks} />
  );
  expect(tree).toMatchSnapshot();
});

test('Sort data by name', () => {
  const tree = shallow(
    <ContactsTable columns={data.metadata.columns} contacts={mocks} />
  );
  tree.setState({ filters: { name: 'mell' } })
  expect(tree).toMatchSnapshot();
})

test('Sort data by age', () => {
  const tree = shallow(
    <ContactsTable columns={data.metadata.columns} contacts={mocks} />
  );
  tree.setState({ filters: { age: '35' } })
  expect(tree).toMatchSnapshot();
})

test('Sort data by city', () => {
  const tree = shallow(
    <ContactsTable columns={data.metadata.columns} contacts={mocks} />
  );
  tree.setState({ filters: { city: 'xin' } })
  expect(tree).toMatchSnapshot();
})

test('Sort data by phone', () => {
  const tree = shallow(
    <ContactsTable columns={data.metadata.columns} contacts={mocks} />
  );
  tree.setState({ filters: { phone: '562' } })
  expect(tree).toMatchSnapshot();
})

test('Remove phone filter', () => {
  const tree = shallow(
    <ContactsTable columns={data.metadata.columns} contacts={mocks} />
  );
  tree.setState({ filters: { phone: '562' } })
  expect(tree).toMatchSnapshot();

  tree.setState({ filters: { phone: '' } })
  expect(tree).toMatchSnapshot();
})
