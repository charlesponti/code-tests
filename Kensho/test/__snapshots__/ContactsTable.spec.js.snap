// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays data 1`] = `
<HTMLTable
  interactive={true}
  striped={true}
>
  <thead>
    <tr>
      <td>
        <div
          style={
            Object {
              "borderBottom": "1px solid black",
              "fontSize": "1.5rem",
              "fontWeight": "bold",
              "marginBottom": "0.5rem",
            }
          }
        >
          Current Filters
        </div>
      </td>
    </tr>
    <tr>
      <HeaderCell
        filter=""
        key="name"
        name="Name"
        onFilter={[Function]}
        type="string"
      />
      <HeaderCell
        filter=""
        key="phone"
        name="Phone Number"
        onFilter={[Function]}
        type="string"
      />
      <HeaderCell
        filter=""
        key="city"
        name="City"
        onFilter={[Function]}
        type="category"
      />
      <HeaderCell
        filter=""
        key="age"
        name="Age"
        onFilter={[Function]}
        type="number"
      />
    </tr>
  </thead>
  <tbody>
    <tr
      key="776-878-8338"
    >
      <td
        key="name"
      >
        Gwen Garrick
      </td>
      <td
        key="phone"
      >
        776-878-8338
      </td>
      <td
        key="city"
      >
        Maquanzhen
      </td>
      <td
        key="age"
      >
        35
      </td>
    </tr>
    <tr
      key="922-694-8628"
    >
      <td
        key="name"
      >
        Carmella Brunker
      </td>
      <td
        key="phone"
      >
        922-694-8628
      </td>
      <td
        key="city"
      >
        General Lavalle
      </td>
      <td
        key="age"
      >
        40
      </td>
    </tr>
    <tr
      key="327-542-4874"
    >
      <td
        key="name"
      >
        Abbot Delahunt
      </td>
      <td
        key="phone"
      >
        327-542-4874
      </td>
      <td
        key="city"
      >
        Xinba
      </td>
      <td
        key="age"
      >
        2
      </td>
    </tr>
    <tr
      key="422-562-2609"
    >
      <td
        key="name"
      >
        Lucine Stratz
      </td>
      <td
        key="phone"
      >
        422-562-2609
      </td>
      <td
        key="city"
      >
        Mlaka pri Kranju
      </td>
      <td
        key="age"
      >
        7
      </td>
    </tr>
    <tr
      key="580-701-9078"
    >
      <td
        key="name"
      >
        Ethe Tomankiewicz
      </td>
      <td
        key="phone"
      >
        580-701-9078
      </td>
      <td
        key="city"
      >
        Kamenický Šenov
      </td>
      <td
        key="age"
      >
        98
      </td>
    </tr>
  </tbody>
</HTMLTable>
`;

exports[`Remove phone filter 1`] = `
<HTMLTable
  interactive={true}
  striped={true}
>
  <thead>
    <tr>
      <td>
        <div
          style={
            Object {
              "borderBottom": "1px solid black",
              "fontSize": "1.5rem",
              "fontWeight": "bold",
              "marginBottom": "0.5rem",
            }
          }
        >
          Current Filters
        </div>
        <div
          key="phone"
          style={
            Object {
              "fontSize": "1rem",
              "marginBottom": "0.5rem",
            }
          }
        >
          <span>
            <b>
              PHONE
            </b>
             : 
            562
          </span>
          <button
            onClick={[Function]}
            style={
              Object {
                "color": "red",
                "fontSize": "1rem",
              }
            }
          >
             Remove 
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <HeaderCell
        key="name"
        name="Name"
        onFilter={[Function]}
        type="string"
      />
      <HeaderCell
        filter="562"
        key="phone"
        name="Phone Number"
        onFilter={[Function]}
        type="string"
      />
      <HeaderCell
        key="city"
        name="City"
        onFilter={[Function]}
        type="category"
      />
      <HeaderCell
        key="age"
        name="Age"
        onFilter={[Function]}
        type="number"
      />
    </tr>
  </thead>
  <tbody>
    <tr
      key="422-562-2609"
    >
      <td
        key="name"
      >
        Lucine Stratz
      </td>
      <td
        key="phone"
      >
        422-562-2609
      </td>
      <td
        key="city"
      >
        Mlaka pri Kranju
      </td>
      <td
        key="age"
      >
        7
      </td>
    </tr>
  </tbody>
</HTMLTable>
`;

exports[`Remove phone filter 2`] = `
<HTMLTable
  interactive={true}
  striped={true}
>
  <thead>
    <tr>
      <td>
        <div
          style={
            Object {
              "borderBottom": "1px solid black",
              "fontSize": "1.5rem",
              "fontWeight": "bold",
              "marginBottom": "0.5rem",
            }
          }
        >
          Current Filters
        </div>
      </td>
    </tr>
    <tr>
      <HeaderCell
        key="name"
        name="Name"
        onFilter={[Function]}
        type="string"
      />
      <HeaderCell
        filter=""
        key="phone"
        name="Phone Number"
        onFilter={[Function]}
        type="string"
      />
      <HeaderCell
        key="city"
        name="City"
        onFilter={[Function]}
        type="category"
      />
      <HeaderCell
        key="age"
        name="Age"
        onFilter={[Function]}
        type="number"
      />
    </tr>
  </thead>
  <tbody>
    <tr
      key="776-878-8338"
    >
      <td
        key="name"
      >
        Gwen Garrick
      </td>
      <td
        key="phone"
      >
        776-878-8338
      </td>
      <td
        key="city"
      >
        Maquanzhen
      </td>
      <td
        key="age"
      >
        35
      </td>
    </tr>
    <tr
      key="922-694-8628"
    >
      <td
        key="name"
      >
        Carmella Brunker
      </td>
      <td
        key="phone"
      >
        922-694-8628
      </td>
      <td
        key="city"
      >
        General Lavalle
      </td>
      <td
        key="age"
      >
        40
      </td>
    </tr>
    <tr
      key="327-542-4874"
    >
      <td
        key="name"
      >
        Abbot Delahunt
      </td>
      <td
        key="phone"
      >
        327-542-4874
      </td>
      <td
        key="city"
      >
        Xinba
      </td>
      <td
        key="age"
      >
        2
      </td>
    </tr>
    <tr
      key="422-562-2609"
    >
      <td
        key="name"
      >
        Lucine Stratz
      </td>
      <td
        key="phone"
      >
        422-562-2609
      </td>
      <td
        key="city"
      >
        Mlaka pri Kranju
      </td>
      <td
        key="age"
      >
        7
      </td>
    </tr>
    <tr
      key="580-701-9078"
    >
      <td
        key="name"
      >
        Ethe Tomankiewicz
      </td>
      <td
        key="phone"
      >
        580-701-9078
      </td>
      <td
        key="city"
      >
        Kamenický Šenov
      </td>
      <td
        key="age"
      >
        98
      </td>
    </tr>
  </tbody>
</HTMLTable>
`;

exports[`Sort data by age 1`] = `
<HTMLTable
  interactive={true}
  striped={true}
>
  <thead>
    <tr>
      <td>
        <div
          style={
            Object {
              "borderBottom": "1px solid black",
              "fontSize": "1.5rem",
              "fontWeight": "bold",
              "marginBottom": "0.5rem",
            }
          }
        >
          Current Filters
        </div>
        <div
          key="age"
          style={
            Object {
              "fontSize": "1rem",
              "marginBottom": "0.5rem",
            }
          }
        >
          <span>
            <b>
              AGE
            </b>
             : 
            35
          </span>
          <button
            onClick={[Function]}
            style={
              Object {
                "color": "red",
                "fontSize": "1rem",
              }
            }
          >
             Remove 
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <HeaderCell
        key="name"
        name="Name"
        onFilter={[Function]}
        type="string"
      />
      <HeaderCell
        key="phone"
        name="Phone Number"
        onFilter={[Function]}
        type="string"
      />
      <HeaderCell
        key="city"
        name="City"
        onFilter={[Function]}
        type="category"
      />
      <HeaderCell
        filter="35"
        key="age"
        name="Age"
        onFilter={[Function]}
        type="number"
      />
    </tr>
  </thead>
  <tbody>
    <tr
      key="776-878-8338"
    >
      <td
        key="name"
      >
        Gwen Garrick
      </td>
      <td
        key="phone"
      >
        776-878-8338
      </td>
      <td
        key="city"
      >
        Maquanzhen
      </td>
      <td
        key="age"
      >
        35
      </td>
    </tr>
  </tbody>
</HTMLTable>
`;

exports[`Sort data by city 1`] = `
<HTMLTable
  interactive={true}
  striped={true}
>
  <thead>
    <tr>
      <td>
        <div
          style={
            Object {
              "borderBottom": "1px solid black",
              "fontSize": "1.5rem",
              "fontWeight": "bold",
              "marginBottom": "0.5rem",
            }
          }
        >
          Current Filters
        </div>
        <div
          key="city"
          style={
            Object {
              "fontSize": "1rem",
              "marginBottom": "0.5rem",
            }
          }
        >
          <span>
            <b>
              CITY
            </b>
             : 
            xin
          </span>
          <button
            onClick={[Function]}
            style={
              Object {
                "color": "red",
                "fontSize": "1rem",
              }
            }
          >
             Remove 
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <HeaderCell
        key="name"
        name="Name"
        onFilter={[Function]}
        type="string"
      />
      <HeaderCell
        key="phone"
        name="Phone Number"
        onFilter={[Function]}
        type="string"
      />
      <HeaderCell
        filter="xin"
        key="city"
        name="City"
        onFilter={[Function]}
        type="category"
      />
      <HeaderCell
        key="age"
        name="Age"
        onFilter={[Function]}
        type="number"
      />
    </tr>
  </thead>
  <tbody>
    <tr
      key="327-542-4874"
    >
      <td
        key="name"
      >
        Abbot Delahunt
      </td>
      <td
        key="phone"
      >
        327-542-4874
      </td>
      <td
        key="city"
      >
        Xinba
      </td>
      <td
        key="age"
      >
        2
      </td>
    </tr>
  </tbody>
</HTMLTable>
`;

exports[`Sort data by name 1`] = `
<HTMLTable
  interactive={true}
  striped={true}
>
  <thead>
    <tr>
      <td>
        <div
          style={
            Object {
              "borderBottom": "1px solid black",
              "fontSize": "1.5rem",
              "fontWeight": "bold",
              "marginBottom": "0.5rem",
            }
          }
        >
          Current Filters
        </div>
        <div
          key="name"
          style={
            Object {
              "fontSize": "1rem",
              "marginBottom": "0.5rem",
            }
          }
        >
          <span>
            <b>
              NAME
            </b>
             : 
            mell
          </span>
          <button
            onClick={[Function]}
            style={
              Object {
                "color": "red",
                "fontSize": "1rem",
              }
            }
          >
             Remove 
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <HeaderCell
        filter="mell"
        key="name"
        name="Name"
        onFilter={[Function]}
        type="string"
      />
      <HeaderCell
        key="phone"
        name="Phone Number"
        onFilter={[Function]}
        type="string"
      />
      <HeaderCell
        key="city"
        name="City"
        onFilter={[Function]}
        type="category"
      />
      <HeaderCell
        key="age"
        name="Age"
        onFilter={[Function]}
        type="number"
      />
    </tr>
  </thead>
  <tbody>
    <tr
      key="922-694-8628"
    >
      <td
        key="name"
      >
        Carmella Brunker
      </td>
      <td
        key="phone"
      >
        922-694-8628
      </td>
      <td
        key="city"
      >
        General Lavalle
      </td>
      <td
        key="age"
      >
        40
      </td>
    </tr>
  </tbody>
</HTMLTable>
`;

exports[`Sort data by phone 1`] = `
<HTMLTable
  interactive={true}
  striped={true}
>
  <thead>
    <tr>
      <td>
        <div
          style={
            Object {
              "borderBottom": "1px solid black",
              "fontSize": "1.5rem",
              "fontWeight": "bold",
              "marginBottom": "0.5rem",
            }
          }
        >
          Current Filters
        </div>
        <div
          key="phone"
          style={
            Object {
              "fontSize": "1rem",
              "marginBottom": "0.5rem",
            }
          }
        >
          <span>
            <b>
              PHONE
            </b>
             : 
            562
          </span>
          <button
            onClick={[Function]}
            style={
              Object {
                "color": "red",
                "fontSize": "1rem",
              }
            }
          >
             Remove 
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <HeaderCell
        key="name"
        name="Name"
        onFilter={[Function]}
        type="string"
      />
      <HeaderCell
        filter="562"
        key="phone"
        name="Phone Number"
        onFilter={[Function]}
        type="string"
      />
      <HeaderCell
        key="city"
        name="City"
        onFilter={[Function]}
        type="category"
      />
      <HeaderCell
        key="age"
        name="Age"
        onFilter={[Function]}
        type="number"
      />
    </tr>
  </thead>
  <tbody>
    <tr
      key="422-562-2609"
    >
      <td
        key="name"
      >
        Lucine Stratz
      </td>
      <td
        key="phone"
      >
        422-562-2609
      </td>
      <td
        key="city"
      >
        Mlaka pri Kranju
      </td>
      <td
        key="age"
      >
        7
      </td>
    </tr>
  </tbody>
</HTMLTable>
`;
